BH/Modules/Item/Item.cpp

```diff
@@ -157,13 +163,372 @@ void GetItemStats()
+		ItemAttributeMap[GetTxtItemCode(pArmor)] = attrs;
+	}
+}
+void GetMiscAttributes()
+{
+	D2ItemDataTbl* pItemDataTables = D2COMMON_10535_DATATBLS_GetItemDataTables();
+
+	for (auto d = 0; d < pItemDataTables->nMiscTxtRecordCount; d++)
+	{
+		ItemsTxt* pMisc = &pItemDataTables->pMisc[d];
+		BYTE stackable = pMisc->bstackable > 0 ? pMisc->bstackable : 0;
+		BYTE useable = pMisc->buseable > 0 ? pMisc->buseable : 0;
+		BYTE throwable = throwableMap[pMisc->nType] > 0 ? throwableMap[pMisc->nType] : 0;
+		unsigned int flags = 0, flags2 = 0;
+
+		std::set<WORD> ancestorTypes;
+		FindAncestorTypes(pMisc->nType, ancestorTypes, parentMap1, parentMap2);
+		FindAncestorTypes(pMisc->wtype2, ancestorTypes, parentMap1, parentMap2);
+
+		if (ancestorTypes.find(ITEM_TYPE_RUNE) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_STACK_RUNE) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_RUNE;
+		}
+
+		// Gem Quality
+		if (ancestorTypes.find(ITEM_TYPE_CHIPPED_GEM) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_CHIPPED;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_FLAWED_GEM) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_FLAWED;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_STANDARD_GEM) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_REGULAR;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_GEN_FLAWLESS_GEM) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_STACK_FLAWLESS) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_FLAWLESS;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_GEN_PERFECT_GEM) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_STACK_PERFECT) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_PERFECT;
+		}
+
+		// Gem Type
+		if (ancestorTypes.find(ITEM_TYPE_AMETHYST) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_AMETHYST) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_AMETHYST;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_DIAMOND) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_DIAMOND) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_DIAMOND;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_EMERALD) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_EMERALD) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_EMERALD;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_RUBY) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_RUBY) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_RUBY;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_SAPPHIRE) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_SAPPHIRE) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_SAPPHIRE;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_TOPAZ) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_TOPAZ) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_TOPAZ;
+		}
+		else if (ancestorTypes.find(ITEM_TYPE_SKULL) != ancestorTypes.end() || ancestorTypes.find(ITEM_TYPE_GEN_SKULL) != ancestorTypes.end()) {
+			flags2 |= ITEM_GROUP_SKULL;
+		}
+
+		ItemAttributes* attrs = new ItemAttributes();
```

Comment:
Just marking these as potential issues, needs the same treatment as the other comments